src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSparkUtils.java

```diff
@@ -80,11 +84,60 @@
                 out.add(new Tuple2<>(pair.key(header), pair));
             }
             return out;
-        }).groupByKey();
+        }).groupByKey(parallelism);
 
         return markPairedEnds(keyedPairs, scoringStrategy, finder, header);
     }
 
+    /**
+     * Like <code>groupByKey</code>, but assumes that values are already sorted by key, so no shuffle is needed,
```

Comment:
The values in this case are already sorted by read name, but the full key as defined by `ReadsKey.keyForRead()` is the concatenation of the read group and the read name:

    public static String keyForRead(final SAMFileHeader header, final GATKRead read) {
        return String.format(
                "%s|%s",
                read.getReadGroup(),
                read.getName());
    }

Does `spanByKey()` account for this? (ie., does it guarantee that two reads with the same name but different read groups will not get grouped together?)